src/lando/api/tests/conftest.py

```diff
@@ -471,3 +500,101 @@ def strptime(cls, date_string, fmt):
             return dates[f"{date_string}"]
 
     return Mockdatetime
+
+
+@pytest.fixture
+def proxy_client(monkeypatch):
+    """A client that bridges tests designed to work with the API.
+
+    Most tests that use the API no longer need to access those endpoints through
+    the API as the data can be fetched directly within the application. This client
+    is a temporary implementation to bridge tests and minimize the number of changes
+    needed to the tests during the porting process.
+
+    This client should be removed and all the tests that depend on it should be
+    reimplemented to not need a response or response-like object.
+    """
+
+    class Response:
+        """Mock response class to satisfy some requirements of tests."""
+
+        # NOTE: The methods tested that rely on this class should be reimplemented
+        # to no longer need the structure of a response to function.
+        def __init__(self, status_code=200, json=None):
+            self.json = json or {}
+            self.status_code = status_code
+            self.content_type = (
+                "application/json"
+                if str(status_code)[0] not in ("4", "5")
+                else "application/problem+json"
+            )
+
+    class _proxy:
```

Comment:
`class TestProxy` would be a nicer name here. If you added a constructor you could pass `monkeypatch` as an argument, and then this could be defined outside the test fixture as well.

Reply:
It's mostly just a desire to avoid nesting things where they could be defined as top-level, making it easier to read, and easier to load into a REPL if we had to debug some problem by hand. I'm not too fussed about it after the addition of the handler methods.